node/src/store/indexed_db_store.rs

```diff
@@ -0,0 +1,533 @@
+use async_trait::async_trait;
+use celestia_types::hash::Hash;
+use celestia_types::ExtendedHeader;
+use rexie::{Direction, Index, KeyRange, ObjectStore, Rexie, TransactionMode};
+use send_wrapper::SendWrapper;
+use serde::{Deserialize, Serialize};
+use serde_wasm_bindgen::{from_value, to_value};
+
+use crate::store::{Result, Store, StoreError};
+
+const DB_VERSION: u32 = 1;
+const HEADER_STORE_NAME: &str = "headers";
+const HASH_INDEX_NAME: &str = "hash";
+const HEIGHT_INDEX_NAME: &str = "height";
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ExtendedHeaderEntry {
+    height: u64,
+    hash: Hash,
+    header: ExtendedHeader,
+}
+
+// SendWrapper usage is safe in wasm because we're running on single thread
+#[derive(Debug)]
+pub struct IndexedDbStore(SendWrapper<Rexie>);
+
+impl IndexedDbStore {
+    pub async fn new_with_name(name: &str) -> rexie::Result<Self> {
```

Comment:
Return `StoreError`. Maybe something like `OpenFailed` to be generic and also adjust the constructor of `SledStore` to use it.

Reply:
Replaced crate specific errors with `StoreError::OpenFailed` with description here and used that in `SledStore`.